rpc/client/client.go

```diff
@@ -388,7 +382,18 @@ func (c *Client) BlockResults(ctx context.Context, height *int64) (*ctypes.Resul
 
 func (c *Client) Commit(ctx context.Context, height *int64) (*ctypes.ResultCommit, error) {
 	// needs block store
-	panic("Commit - not implemented!")
+	heightValue := c.normalizeHeight(height)
+	block, err := c.node.Store.LoadBlock(heightValue)
```

Comment:
Instead of using `LoadBlock` and the `block.LastCommit` you should use `c.node.Store.LoadCommit`. Your method is returning wrong value (previous commit, not commit for block at given height).

You can take a look at [Tendermint implementation](https://github.com/celestiaorg/celestia-core/blob/b00cac936825b61c1f60fd888794d1d77339f0b9/rpc/core/blocks.go#L110).